wolfcrypt/src/port/maxim/max3266x.c

```diff
@@ -0,0 +1,921 @@
+#if defined(WOLFSSL_MAX32665) ||  defined(WOLFSSL_MAX32666)
+#warning "Inside max3266x.c define"
+
+
+#ifdef NO_INLINE
+    #include <wolfssl/wolfcrypt/misc.h>
+#else
+    #define WOLFSSL_MISC_INCLUDED
+    #include <wolfcrypt/src/misc.c>
+#endif
+
+#include <stdint.h>
+#include <wolfssl/internal.h>
+#include <wolfssl/wolfcrypt/error-crypt.h>
+#include <wolfssl/wolfcrypt/logging.h>
+#include <wolfcrypt/src/misc.c>
+#include <wolfssl/wolfcrypt/port/maxim/max3266x.h>
+#include <wolfssl/wolfcrypt/types.h>
+
+#if defined(USE_FAST_MATH) && defined(WOLFSSL_SP_MATH_ALL)
+    #error Cannot Do Both FAST MATH and SP MATH ALL with MAX3266X
+#elif defined(USE_FAST_MATH)
+    #error  MXC Not Compatible with Fast Math 
+    #warning Fast Math is being used
+    #include <wolfssl/wolfcrypt/tfm.h>
+    #define MXC_WORD_SIZE               DIGIT_BIT
+#elif defined(WOLFSSL_SP_MATH_ALL)
+    #warning Assuming SP MATH ALL is being used
+    #include <wolfssl/wolfcrypt/sp_int.h>
+    #define MXC_WORD_SIZE               SP_WORD_SIZE
+#endif
+
+#if defined(MAX3266X_MATH)
+    #if defined(USE_FAST_MATH)
+        #define mxc_mod         fp_mod
+        #define mxc_addmod      fp_addmod
+        #define mxc_submod      fp_submod
+        #define mxc_mulmod      fp_mulmod
+        #define mxc_exptmod     fp_exptmod
+        #define mxc_sqrmod      fp_sqrmod
+    #elif defined(WOLFSSL_SP_MATH_ALL)
+        #define mxc_mod         sp_mod
+        #define mxc_addmod      sp_addmod
+        #define mxc_submod      sp_submod
+        #define mxc_mulmod      sp_mulmod
+        #define mxc_exptmod     sp_exptmod
+        #define mxc_sqrmod      sp_sqrmod
+    #else
+        #error Need USE_FAST_MATH OR WOLFSSL_SP_MATH_ALL
+    #endif
+#endif
+
+int wc_MXC_TPU_Init(void)
+{
+    /* Initialize the TPU device */
+    if(MXC_TPU_Init(MXC_SYS_PERIPH_CLOCK_TRNG) != EXIT_SUCCESS){
+        WOLFSSL_MSG("Device did not initialize");
+        return RNG_FAILURE_E;
+    }
+    return EXIT_SUCCESS;
+}
+
+int wc_MXC_TPU_Shutdown(void)
+{
+    /* Shutdown the TPU device */
+    if(MXC_TPU_Shutdown(MXC_SYS_PERIPH_CLOCK_TRNG) != EXIT_SUCCESS){
+        WOLFSSL_MSG("Device did not shutdown");
+        return RNG_FAILURE_E;
+    }
+    return EXIT_SUCCESS;
+}
+
+
+/* Convert Error Codes Correctly */
+/* TODO: Convert to correct wolfCrypt Codes */
+/* TODO: Add wolfssl Message Statements to report HW issue on bad return */
+int wc_MXC_error(int *ret)
+{
+    switch(*ret){
+        case(E_SUCCESS):
+            return E_SUCCESS;
+
+        case(E_NULL_PTR):
+            return E_NULL_PTR;
+
+        case(E_INVALID): /* Process Failed */
+            return E_INVALID;
+
+        case(E_BAD_PARAM):
+            return BAD_FUNC_ARG;
+
+        case(E_BAD_STATE): 
+            return E_BAD_STATE;
+
+        default:
+            *ret = WC_HW_E; /* If something else return HW Error */
+            return *ret;
+    }
+}
+
+
+#if defined(MAX3266X_RNG)
+
+    /* Use this RNG_FAILURE_E for RNG Errors*/
```

Comment:
Reduce indent for entire function please. Same for others. I don't want functions indented due to if macros.